Shared/Samples/Authenticate with PKI certificate/AuthenticateWithPKICertificateView.swift

```diff
@@ -0,0 +1,188 @@
+
+     var id: ObjectIdentifier {
+         ObjectIdentifier(portalItem)
+     }
+ }
+
+ extension AuthenticateWithPKICertificateView {
+     @MainActor
+     class Model: ObservableObject {
+         /// The authenticator to handle authentication challenges.
+         let authenticator = Authenticator()
+
+         /// The URL string entered by the user.
+         @Published var portalURLString = ""
+
+         /// The fetched portal content.
+         @Published var portalContent: Result<PortalQueryResultSet<PortalItem>, Error>?
+
+         /// A Boolean value indicating if a portal connection is in progress.
+         @Published var isConnecting = false
+
+         /// The selected item.
+         @Published fileprivate var selectedItem: SelectedItem?
+
+         /// The URL of the portal.
+         var portalURL: URL? { URL(string: portalURLString) }
+
+         init() {
+             setupAuthenticator()
+         }
+
+         /// Connects to the portal and finds a batch of webmaps.
+         func connectToPortal() async {
+             precondition(portalURL != nil)
+
+             isConnecting = true
+             defer { isConnecting = false }
+
+             do {
+                 let portal = Portal(url: portalURL!)
+                 try await portal.load()
+                 let results = try await portal.findItems(queryParameters: .items(ofKinds: [.webMap]))
+                 portalContent = .success(results)
+             } catch {
+                 portalContent = .failure(error)
+             }
+         }
+
+         /// Sets up the authenticator to handle challenges.
+         private func setupAuthenticator() {
+             // Setting the challenge handlers here when the model is created so user is prompted to enter
+             // credentials every time trying the sample. In real world applications, set challenge
+             // handlers at the start of the application.
+
+             // Sets authenticator as ArcGIS and Network challenge handlers to handle authentication
+             // challenges.
+             ArcGISEnvironment.authenticationManager.handleChallenges(using: authenticator)
+
+             // In your application you may want to uncomment this code to persist
+             // credentials in the keychain.
+             // setupPersistentCredentialStorage()
+         }
+
+         /// Stops the authenticator from handling the challenges and clears credentials.
+         nonisolated func teardownAuthenticator() async {
+             // Resets challenge handlers.
+             ArcGISEnvironment.authenticationManager.handleChallenges(using: nil)
+
+             // In your application, code may need to run at a different
+             // point in time based on the workflow desired. For example, it
+             // might make sense to remove credentials when the user taps
+             // a "sign out" button.
+             await ArcGISEnvironment.authenticationManager.revokeOAuthTokens()
```

Comment:
This is not needed for PKI and IWA samples as we are not expecting to use `OAuth`. I missed to review this in IWA sample. I'm ok if we want to keep this for both samples.

Reply:
I purposefully left it in the IWA sample as I felt that this code might just be copied and used for other Auth types. Also for PKI, you might actually have dual authentication.

Reply:
Ok, all auth samples should do the same. 👍